frontend/src/components/TeamProject/TeamProjectSlice.ts

```diff
@@ -0,0 +1,152 @@
+    parentProjectId: "loading...",
+    projectName: "loading...",
+    projectUrl: "loading...",
+    description: "loading...",
+    mentor: "loading...",
+    referenceProjectName: "loading...",
+    sectionName: "loading..."
+  }
+};
+
+export const teamProjectSlice = createSlice({
+  name: 'teamProject',
+  initialState,
+  reducers: {
+    projectOperation: state => {
+      state.loading = true;
+    },
+    projectOperationSuccess: (state, { payload }) => {
+      state.project = payload;
+      state.loading = false;
+      state.hasErrors = false;
+    },
+    projectDeleteSuccess: (state) => {
+      state.loading = false;
+      state.hasErrors = false;
+    },
+    projectOperationFailure: state => {
+      state.loading = true;
+      state.hasErrors = true;
+    },
+    switchEditMode: state => {
+      if (!state.loading || state.projectEditMode === true) state.projectEditMode = !state.projectEditMode;
+    },
+    switchDeleteMode: state => {
+      if (!state.loading || state.projectDeleteMode === true) state.projectDeleteMode = !state.projectDeleteMode;
+    }
+  }
+});
+
+export const { 
+  projectOperation, 
+  projectOperationSuccess, 
+  projectDeleteSuccess,
+  projectOperationFailure, 
+  switchEditMode, 
+  switchDeleteMode 
+} = teamProjectSlice.actions;
+
+// if you want, add selectors here, change the one below, remember to register reducer in store.ts
+export const selectTeamProjects = (state: RootState) => state.teamProjects;
+
+export default teamProjectSlice.reducer;
+
+const api = new BaseService();
+
+export function getProjectById(id: string, token: string) {
+  return async (dispatch: Dispatch) => {
+    dispatch(projectOperation())
+    try {
+      const responseProject = await api.get(`/teams/projects/${id}`);
+      const data = await responseProject.data;
+
+      const responseTeam = await api.get(`teams/${data.teamId}`)
+      const dataTeam = await responseTeam.data;
+      data.mentor = `${dataTeam.mentor.name} ${dataTeam.mentor.surname}`;
+
+      const responseParentProject = await api.get(`/projects/${data.parentProjectId}`);
+      const dataParentProject = await responseParentProject.data;
+      data.referenceProjectName = dataParentProject.projectName;
+
+      try {
+        const responseSection = await api.get(`/sections/${dataParentProject.sectionId}`);
+        const dataSection = await responseSection.data;
+        data.sectionName = dataSection.name;
+      }
+      catch (error){
+        console.log('Section',error.message)
+      }
+
+      dispatch(projectOperationSuccess(data));
+      dispatch(switchEditMode);
+    } catch (error) {
+      dispatch(projectOperationFailure());
+    }
+  }
+}
+
+export function saveProjectById(project: Object, id: string, token: string) {
+  return async (dispatch: Dispatch) => {
+    dispatch(projectOperation());
+
+    const putConfig = {
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8'
+      }
+    }
```

Comment:
In my opinion this header should be defined in baseService.ts file.

Reply:
done